boot/boot.js

```diff
@@ -287,6 +287,92 @@ $tw.utils.decodeURIComponentSafe = function(s) {
 	return v;
 };
 
+/*
+Helpers for encoding/decoding URIs
+*/
+
+/*
+The character that will substitute for a space in the URL
+*/
+var SPACE = "_";
+/*
+The character that will separate out the list elements in the URL
+*/
+var CONJUNCTION = ";";
+/*
+Those of the allowed url characters claimed by TW
+*/
+var claimed= [SPACE, ":", CONJUNCTION];
+/*
+Non-alphanumeric characters allowed in a URL fragment
+More information at https://www.rfc-editor.org/rfc/rfc3986#appendix-A
+*/
+var pchar = "-._~!$&'()*+,;=:@/?";
+/*
+The subset of the pchars we will not percent-encode in permalinks/permaviews
+*/
+var substitutes = pchar.split('').filter(c => claimed.indexOf(c) == -1);
+/*
+A regex to match the percent-encoded characters we will want to replace.
+Something similar to the following, depending on SPACE and CONJUNCTION
+    /(%2D|%2E|%7E|%21|%24|%26|%27|%28|%29|%2A|%2B|%3B|%3D|%40|%2F|%3F)/g
+*/
+var charMatch = new RegExp(
+	"(" + substitutes.map(c =>  "%" + c.charCodeAt(0).toString(16).toUpperCase()).join('|') + ')', 
+	"g"
+)
+/*
+A regex to match the SPACE character
+*/
+var spaceMatch = new RegExp("(\\" + SPACE + ")", "g")
+/*
+An object mapping the percent encodings back to their source characters
+*/
+var pctCharMap = substitutes.reduce(function (a, c) {
+	a['%' + c.charCodeAt(0).toString(16).toUpperCase()] = c
+	return a
+}, {});
+
+/*
+Convert a URI List Component encoded string (with `+` as an allowed replacement 
```

Comment:
IMO you should use `(with SPACE as an allowed ...` instead of `+` -- Check all comments

Reply:
Do you mean

> (with `+` as an allowed replacement for SPACE)

This is decoding `Foo+Bar` to `Foo Bar`.